static/install-linux.sh

```diff
@@ -58,80 +58,141 @@ fi
+        icon_img_path="$try_icon_path"
+    fi
+fi
 if [ ! -f "$icon_img_path" ]; then
     echo "WARNING: Missing '$icon_img_path' (shortcut will have no icon image)"
 fi
 shortcut_name=com.blackflux.voxelshop.desktop
 shortcut_path=$install_src/share/applications/$shortcut_name
-
+if [ ! -f "$shortcut_path" ]; then
+    try_shortcut_path="$SCRIPT_DIR/share/applications/$shortcut_name"
+    if [ -f "$try_shortcut_path" ]; then
+        echo "INFO: $shortcut_path wasn't present, so $try_shortcut_path will be used instead."
+        shortcut_path="$try_shortcut_path"
+    fi
+fi
 # endregion hardcoded settings
 
 if [ ! -d $PREFIX/lib ]; then
-    mkdir -p $PREFIX/lib || customDie "Cannot create $PREFIX/lib"
+    mkdir -p $PREFIX/lib || customExit "Cannot create $PREFIX/lib"
 fi
 
 if [ ! -d $PREFIX/bin ]; then
-    mkdir -p $PREFIX/bin || customDie "Cannot create $PREFIX/bin"
+    mkdir -p $PREFIX/bin || customExit "Cannot create $PREFIX/bin"
 fi
 
 if [ ! -d "$dest_dir_path/lib" ]; then
     msg="Cannot create '$dest_dir_path/lib'"
-    mkdir -p $dest_dir_path/lib || customDie "$msg"
+    mkdir -p $dest_dir_path/lib || customExit "$msg"
 fi
 
 if [ ! -d "$dest_dir_path/share/applications" ]; then
     msg="Cannot create '$dest_dir_path/share/applications'"
-    mkdir -p $dest_dir_path/share/applications || customDie "$msg"
+    mkdir -p $dest_dir_path/share/applications || customExit "$msg"
 fi
 install_log_name=install.log
 install_log=$dest_dir_path/$install_log_name
 if [ -f "$install_log" ]; then
     # stopping in this case avoids creating a faulty install_log that
     # may contain files the user added to the directory.
-    customDie "The program is already installed. Run uninstall-linux.sh or delete $dest_dir_path before trying to reinstall."
+    customExit "The program is already installed. Run uninstall-linux.sh or delete $dest_dir_path before trying to reinstall."
+fi
+
+if [ -d "$dest_dir_path" ]; then
+    echo "WARNING: \"$dest_dir_path\" already exists but there is no install log. The directory will be upgraded (any prepackaged files that remain will be overwritten)."
 fi
-cp -f $install_src/* $dest_dir_path/
+
+# cp -f $install_src/* $dest_dir_path/
+cp -R $install_src $programs_path/
+# ^ dest_dir_path can't be used for cp -R, otherwise if it exists, the
+#   folder will be copied inside the dest instead of as the dest.
+
 find $dest_dir_path | grep -v "$install_log_name" > $install_log
 if [ ! -f "$install_log" ]; then
-    customDie "Cancelling install since cannot create $install_log."
+    customExit "Cancelling install since cannot create $install_log."
+fi
+
+if [ ! -d "$dest_dir_path/lib" ]; then
+    customExit "lib wasn't copied, so install could not complete."
 fi
-cp -f $install_src/lib/* $dest_dir_path/lib/
-find $dest_dir_path | grep -v "$install_log_name" >> $install_log
+# cp -f $install_src/lib/* $dest_dir_path/lib/
+# find $dest_dir_path | grep -v "$install_log_name" >> $install_log
```

Comment:
Should these two commented out lines be removed?

Reply:
Done